### DISCUSSION TITLE
Limon's motion controller + head tracking

### DISCUSSION THREAD

**Discussion by ErikWTN1** [EdgeTX/edgetx#5114] [Ideas]
Limon created a motion controller to fly fpv planes and quads similar to the DJI motion controller, but it requires connecting a flight controller to the radio via the trainer port. For FPV planes, it would be amazing if you could use this controller together with head tracking, which also uses the trainer port. Limon suggested to submit a feature request to "support SBUS + PPM trainer ports at the same time." I don't fully understand how this works so I'm hoping others will be interested in this and help to formulate a coherent feature request.

https://www.youtube.com/watch?v=fegQ7T-UbE8

**Comment by pfeerick** [comment]
If I understand how Limon is using the FC here, IMO if the I2C pins are available for use on the MT12, it would be better to allow for an IMU like https://www.adafruit.com/product/4480 (as is possible with the TX16S) and then it would just appear as another two mixer sources (TltX and TltY). At the moment part of the problem is that SBUS *is* trainer, so you're wanting two trainer sources... which doesn't really compute for most scenarios. Would be a lot cheaper, as well as allowing you to stuff it inside the handset. Although if you have a old FC knocking around, that is _technically_ cheaper :zany_face:

**Reply by ErikWTN1** [reply]
That's an interesting idea! Definitely cheaper and mounting internally would be nice. I don't have an MT12 yet so I don't know if the I2C pins are available.

**Reply by limonspb** [reply]
Well, I have MT12 and still don't know if it has I2C lol.
@pfeerick why two channels and no yaw I guess?

I am also curios about gyro/accel update rate in edge TX and how quicky the sensor fusion filter converges?

**Reply by pfeerick** [reply]
Pretty much all, if not all STM32 MCUs, have I2C... the question will simply be one of 1) since most pins are also multi function, have they been used for something else, and 2) if it has not been used for something else, is it something you can actually connect to :hand_over_mouth: You probably *don't* want to have to solder directly onto the MCU pins :construction: 

I think because until now that has all that has been needed, for the like 2 handsets that actually have IMUs as standard, and as a novelty addon for the TX16S. i.e. was enough to control your pitch and roll while you grabbing that beer out of the esky :zany_face: 

IIRC there is another radio that will come with it as standard, so may see more use cases for it arise, as well as  encouraging more development in this space in the near future.

**Reply by pfeerick** [reply]
Well, there goes that idea... the I2C pins are in use by the internal RF module... so you can have either IMU or internal RF, not both! :rofl: 

The PPM (trainer) and UART pins don't overlap physically, so unless there is an issue with timers, etc, it seems conceivable that both could run both together.

**Reply by pfeerick** [reply]
And then this got announced :grin: (knew it was in the pipeline, just not when going to release)
https://www.radiomasterrc.com/products/mt12-gyro-control-module

**Reply by limonspb** [reply]
> And then this got announced 😁 (knew it was in the pipeline, just not when going to release) https://www.radiomasterrc.com/products/mt12-gyro-control-module

Just watch out it's two axis. Which is good for wings, not drone 😀

**Reply by ErikWTN1** [reply]
Just saw that in my FB feed. Too bad it's just two axis. Still pretty cool though.
